internal/boxcli/setup.go

```diff
@@ -44,28 +44,30 @@ func runInstallNixCmd(cmd *cobra.Command) error {
 		)
 		return nil
 	}
-	return nix.Install(cmd.ErrOrStderr(), nixDaemonFlagVal(cmd))
+	return nix.Install(cmd.ErrOrStderr(), nixDaemonFlagVal(cmd)())
 }
 
 func ensureNixInstalled(cmd *cobra.Command, _args []string) error {
 	return nix.EnsureNixInstalled(cmd.ErrOrStderr(), nixDaemonFlagVal(cmd))
 }
 
-func nixDaemonFlagVal(cmd *cobra.Command) *bool {
-	if !cmd.Flags().Changed(nixDaemonFlag) {
-		if os.Geteuid() == 0 {
-			ux.Fwarning(
-				cmd.ErrOrStderr(),
-				"Running as root. Installing Nix in multi-user mode.\n",
-			)
-			return lo.ToPtr(true)
+func nixDaemonFlagVal(cmd *cobra.Command) func() *bool {
```

Comment:
its a bit unusual to return a `func`, so could you write a comment explaining the motivation: 
we don't want to print the multi-user warning unless this function is actually being used by nix.Install...

its won't be obvious to the casual reader, especially without the context of this PR